Xamarin.PropertyEditing/OrderedDictionary.cs

```diff
@@ -333,7 +377,11 @@ public bool Remove (TKey key)
 		public void RemoveAt (int index)
 		{
 			TKey key = this.keyOrder[index];
-			Remove (key);
+
+			this.keyOrder.RemoveAt (index);
+			this.dict.TryRemove (key, out var value);
```

Comment:
I'd defend in depth against the entry not being in the dictionary with an `Assert`.

Reply:
The try remove isn't defensive, it's to also grab the value for use in the `INCC` event.

Reply:
Sure, what I'm saying is that it should never happen, so I'd `Assert` on the result of the `TryRemove`. Also, there may be a possibility for inconsistent state here as well if that happened.